pennylane_qiskit/converter.py

```diff
@@ -157,23 +246,80 @@ def load(quantum_circuit: QuantumCircuit, measurements=None):
+    def _function(*args, params: dict = None, wires: list = None, **kwargs):
         """Returns a PennyLane template created based on the input QuantumCircuit.
         Warnings are created for each of the QuantumCircuit instructions that were
         not incorporated in the PennyLane template.
 
+        If the QuantumCircuit contains unbound parameters, this function must be passed
+        parameter values. These can be passed as positional arguments (provided in the same
+        order that they are stored on the QuantumCircuit used to generate this function),
+        keyword arguments with keywords matching the parameter names, or as a single dictionary
+        with the parameters (not parameter names) as keys.
+
         Args:
-            params (dict): specifies the parameters that need to be bound in the QuantumCircuit
+            *args: positional arguments defining the value of the parameters that need to be assigned
+            in the QuantumCircuit
+
+        Kwargs:
+            **kwargs: keyword arguments matching the names of the parameters beind defined
+            params (dict): a dictionary with Parameter keys, specifying the parameters that need
+                to be bound in the QuantumCircuit
             wires (Sequence[int] or int): The wires the converted template acts on.
                 Note that if the original QuantumCircuit acted on :math:`N` qubits,
                 then this must be a list of length :math:`N`.
 
         Returns:
             function: the new PennyLane template
+
+        .. warning::
+
+            Because this function takes ``params`` and ``wires`` as arguments in addition to taking
+            any args and kwargs to set parameter values, the ``params`` and ``wires`` arguments are
+            reserved, and cannot be parameter names in the QuantumCircuit.
+
+        **Example**
+
+        This function was created by doing something like:
+
+        .. code-block:: python
+
+            from qiskit.circuit import Parameter, QuantumCircuit
+            import pennylane as qml
+
+            a = Parameter("alpha")
+            b = Parameter("beta")
+            c = Parameter("gamma")
+
+            qc = QuantumCircuit(2, 2)
+            qc.rx(a, 0)
+            qc.rx(b * c, 1)
+
+            this_function = qml.from_qiskit(qc, measurements=[qml.PauliZ(0), qml.PauliZ(1)])
+
+        For the circuit above, based on ``Parameters`` with names 'alpha', 'beta' and 'gamma', all of the following are
+        valid inputs to set the Parameter values for this template:
+
+        .. code-block:: python
+
+            # positional arguments passed in alphabetical order (alpha, beta, gamma)
+            circuit = this_function(np.pi, 0.5, np.pi)
+
+            # kwargs are parameter names
+            circuit = this_function(alpha=np.pi, beta=0.5, gamma=np.pi)
+
+            # kwargs and args can be combined
+            circuit = this_function(np.pi, np.pi, beta=0.5)
```

Comment:
If we have a function `f(a, b, c)`, we can't do `f(1.0, 1.0, b=2.0)`, since `b` is the second positional argument.  But I'm fine with leaving this in as is.